strapi/strapi
# fix(chore): replace use-context-selector context with React Context

### What does it do?

Fix the issue we had after registration because the token was not updated correctly when we are redirected to the usecase page

### Why is it needed?

With use-context-selector context when we save the token in the registration handler, using the setToken coming from the useAuth hook, the value is not immediately updated to be used in the PrivateRoute component so the UseCasePage page is not served

### How to test it?

- clean the db
- create a new admin and flag and enable telemetry
- check if the user is redirected to the /usecase page instead of the /admin

### Related issue(s)/PR(s)

Related issue:  https://github.com/strapi/strapi/issues/20345
